### PR TITLE
Es mapping limit split index

### PR DESCRIPTION
**Pull request by matthopson**

### CODE REVIEW

**Review by codyseibert — APPROVED** [review]

**Comment by ericsorenson** [comment]
Would it have been possible to split this into smaller PRs?

**Comment by matthopson** [comment]
@ericsorenson I don't think so. The bulk of this PR is tests and seed data updates.

File: shared/src/business/useCases/processStreamRecordsInteractor.js [hunk 1]
```diff
@@ -96,7 +99,7 @@ const filterRecords = async ({ applicationContext, records }) => {
         ...caseRecord.dynamodb.NewImage,
         docketRecord: undefined,
         documents: undefined,
-        entityName: undefined,
+        entityName: { S: 'Document' },
```
Comment (kkoskelin) [inline shared/src/business/useCases/processStreamRecordsInteractor.js:102]: Can you tell me more about this change?

Comment (matthopson) [reply shared/src/business/useCases/processStreamRecordsInteractor.js:102]: For documents on the case to be processed into the right index, they need an entityName. This code was already doing some "exceptional" things with those documents. I'm just updating it to ensure it has the entityName property.

File: web-api/delete-elasticsearch-index.js [hunk 2]
```diff
@@ -29,17 +29,20 @@
...
-  try {
-    const indexExists = await searchClientCache.indices.exists({
-      body: {},
-      index: 'efcms',
-    });
-    if (indexExists) {
-      searchClientCache.indices.delete({
-        index: 'efcms',
+  // TODO: DRY up index names array
+  ['efcms', 'efcms-case', 'efcms-document', 'efcms-user'].map(async index => {
```
Comment (kkoskelin) [inline web-api/delete-elasticsearch-index.js]: 🚫 warning!  Although you've defined this function as "async", the native `map` implementation will NOT wait for it (nor will a `foreach`).  You'll need to write this as a more traditional loop (e.g. `for (const index in indexes) { await expr; }`

File: web-api/elasticsearch/elasticsearch-index-settings.js [hunk 3]
```diff
@@ -31,25 +29,27 @@
...
-      searchClientCache.indices.create({
-        body: {
-          settings,
-        },
-        index,
-      });
-    } else {
-      searchClientCache.indices.putSettings({
-        body: { 'index.mapping.total_fields.limit': '4000' },
+  ['efcms', 'efcms-case', 'efcms-document', 'efcms-user'].map(async index => {
```
Comment (kkoskelin) [inline web-api/elasticsearch/elasticsearch-index-settings.js]: 🚫 same caveat applies here -- async won't be applied.

File: web-api/src/applicationContext.js [hunk 4]
```diff
@@ -843,6 +854,11 @@ let searchClientCache;
 
 const entitiesByName = {
   Case: Case,
+  Document: Document,
+  IrsPractitioner: IrsPractitioner,
+  Practitioner: Practitioner,
+  PrivatePractitioner: PrivatePractitioner,
+  User: User,
```
Comment (kkoskelin) [inline web-api/src/applicationContext.js]: nbd, nit: any reason to not use the property shorthand of `Document` instead of `Document: Document`, etc?

**Review by kkoskelin — CHANGES_REQUESTED** [review]
A few questions and a few probably change needed.  But I see where this is going, and I really like it.  Nice work.

File: web-api/delete-elasticsearch-index.js (hunk 2 above)
Comment (matthopson) [reply web-api/delete-elasticsearch-index.js]: Addressed

File: web-api/elasticsearch/elasticsearch-index-settings.js (hunk 3 above)
Comment (matthopson) [reply web-api/elasticsearch/elasticsearch-index-settings.js]: Addressed

File: web-api/src/applicationContext.js (hunk 4 above)
Comment (matthopson) [reply web-api/src/applicationContext.js]: Addressed

File: web-api/delete-elasticsearch-index.js [hunk 5]
```diff
@@ -29,17 +29,22 @@
...
-    if (indexExists) {
-      searchClientCache.indices.delete({
-        index: 'efcms',
-      });
-    }
-  } catch (e) {
-    console.log(e);
-  }
+  // TODO: DRY up index names array
+  await Promise.all(
```
Comment (kkoskelin) [inline web-api/delete-elasticsearch-index.js:33]: as long as it's not a problem to issue and wait on multiple deletions, this looks like a great solution.

**Review by kkoskelin — APPROVED** [review]

**Pull request merged**